### PR TITLE
New api-version 2020-12-01

### PR DESCRIPTION
**Pull request by pvbchak**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [x] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by chiragg4u** [comment]
Please go through the contribution checklist added as part of the review and mark them completed. This is a requirement before review can be continued.

**Comment by panchagnula** [comment]
@pvbchak thanks a lot for doing this - just to make sure are  all changes here  are part of Ant92/Ant93 - asking because you are merging this to master - & would like to avoid this accidentally getting merged & picked up for an SDK generation; before the changes are deployed. Thanks!

File: specification/web/resource-manager/Microsoft.CertificateRegistration/stable/2020-12-01/examples/ListOperations.json
Comment (j5lim) [inline specification/web/resource-manager/Microsoft.CertificateRegistration/stable/2020-12-01/examples/ListOperations.json:3]: >2020-09-01 [](start = 20, length = 10)

Should be 2020-12-01

File: specification/web/resource-manager/Microsoft.DomainRegistration/stable/2020-12-01/examples/GetTopLevelDomain.json
Comment (j5lim) [inline specification/web/resource-manager/Microsoft.DomainRegistration/stable/2020-12-01/examples/GetTopLevelDomain.json:4]: >2020-09-01 [](start = 20, length = 10)

Please fix the version in all examples.

**Comment by j5lim** [comment]
>         "summary": "Get the health details of an App Service Environment.",

This is a list operation. This should return an array of 'health' resources.

---
Refers to: specification/web/resource-manager/Microsoft.Web/stable/2020-12-01/AppServiceEnvironments.json:640 in 8f042f2. [](commit_id = 8f042f20612fdd7710def7f55d936c89b4e33cdc, deletion_comment = False)

**Comment by j5lim** [comment]
>     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/hostingEnvironments/{name}/health": {

Resource type name should be plural.
[https://armwiki.azurewebsites.net/api_contracts/guidelines/openapi.html?q=plural#oapi010-resource-type-names-must-be-plural](https://armwiki.azurewebsites.net/api_contracts/guidelines/openapi.html?q=plural#oapi010-resource-type-names-must-be-plural)

---
Refers to: specification/web/resource-manager/Microsoft.Web/stable/2020-12-01/AppServiceEnvironments.json:635 in 8f042f2. [](commit_id = 8f042f20612fdd7710def7f55d936c89b4e33cdc, deletion_comment = False)

**Comment by j5lim** [comment]
>             "$ref": "./examples/AseHealth.json"

Please add an example for list operation.

---
Refers to: specification/web/resource-manager/Microsoft.Web/stable/2020-12-01/AppServiceEnvironments.json:677 in 8f042f2. [](commit_id = 8f042f20612fdd7710def7f55d936c89b4e33cdc, deletion_comment = False)

**Comment by j5lim** [comment]
>           "202": {

201 is the recommended async pattern for PUT.
[https://armwiki.azurewebsites.net/api_contracts/guidelines/rpc.html#rpc0025-201-is-the-recommended-async-pattern](https://armwiki.azurewebsites.net/api_contracts/guidelines/rpc.html#rpc0025-201-is-the-recommended-async-pattern)

---
Refers to: specification/web/resource-manager/Microsoft.Web/stable/2020-12-01/AppServiceEnvironments.json:1398 in 8f042f2. [](commit_id = 8f042f20612fdd7710def7f55d936c89b4e33cdc, deletion_comment = False)

**Comment by j5lim** [comment]
>         }

Please mark all read-only properties.

---
Refers to: specification/web/resource-manager/Microsoft.Web/stable/2020-12-01/AppServiceEnvironments.json:2453 in 8f042f2. [](commit_id = 8f042f20612fdd7710def7f55d936c89b4e33cdc, deletion_comment = False)

**Comment by j5lim** [comment]
>       }

please mark any read-only properties as read-only.

---
Refers to: specification/web/resource-manager/Microsoft.Web/stable/2020-12-01/AppServiceEnvironments.json:2613 in 8f042f2. [](commit_id = 8f042f20612fdd7710def7f55d936c89b4e33cdc, deletion_comment = False)

File: specification/web/resource-manager/Microsoft.Web/stable/2020-12-01/Certificates.json
Comment (j5lim) [inline specification/web/resource-manager/Microsoft.Web/stable/2020-12-01/Certificates.json:533]: >"password" [](start = 12, length = 10)

Please annotate with x-ms-secret.
[https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html#common-issues-found-in-review](https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html#common-issues-found-in-review)

**Comment by j5lim** [comment]
>           "description": "Access key for the storage account.",

Please annotatge with x-ms-secret: true.
[https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html#common-issues-found-in-review](https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html#common-issues-found-in-review)

---
Refers to: specification/web/resource-manager/Microsoft.Web/stable/2020-12-01/CommonDefinitions.json:462 in 8f042f2. [](commit_id = 8f042f20612fdd7710def7f55d936c89b4e33cdc, deletion_comment = False)

File: specification/web/resource-manager/Microsoft.Web/stable/2020-12-01/Global.json
Comment (j5lim) [inline specification/web/resource-manager/Microsoft.Web/stable/2020-12-01/Global.json:109]: >/subscriptions/{subscriptionId}/providers/Microsoft.Web/locations/{location}/operationResults":  [](start = 5, length = 96)

Do you really need an api to list all async operations? Usually the async pattern is that the client just need to call the uri in the location or azure-asyncOperation header to track the async operation.
[https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#creating-or-updating-resources-asynchronously](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#creating-or-updating-resources-asynchronously)

File: specification/web/resource-manager/Microsoft.Web/stable/2020-12-01/Global.json
Comment (j5lim) [inline specification/web/resource-manager/Microsoft.Web/stable/2020-12-01/Global.json:207]: >subscriptions/{subscriptionId}/providers/Microsoft.Web/locations/{location}/operations/{operationId} [](start = 6, length = 100)

Why do you need this given the above api can track an async operation?

File: specification/web/resource-manager/Microsoft.Web/stable/2020-12-01/Global.json
Comment (j5lim) [inline specification/web/resource-manager/Microsoft.Web/stable/2020-12-01/Global.json:277]: >ArmOperationResponseError" [](start = 5, length = 26)

Consider referencing the common type json.

**Comment by j5lim** [comment]
>               "$ref": "./CommonDefinitions.json#/definitions/StringDictionary"

Can you please define the schema? Otherwise platform feature like Azure policy won't work for this resource type.
[https://armwiki.azurewebsites.net/api_contracts/guidelines/openapi.html#oapi016-string-types-that-should-be-objects](https://armwiki.azurewebsites.net/api_contracts/guidelines/openapi.html#oapi016-string-types-that-should-be-objects)

---
Refers to: specification/web/resource-manager/Microsoft.Web/stable/2020-12-01/StaticSites.json:740 in 8f042f2. [](commit_id = 8f042f20612fdd7710def7f55d936c89b4e33cdc, deletion_comment = False)

**Comment by j5lim** [comment]
>           "202": {

Please set x-ms-long-running-operation to true if it's async.
[https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-long-running-operation](https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-long-running-operation)

---
Refers to: specification/web/resource-manager/Microsoft.Web/stable/2020-12-01/StaticSites.json:943 in 8f042f2. [](commit_id = 8f042f20612fdd7710def7f55d936c89b4e33cdc, deletion_comment = False)

**Comment by j5lim** [comment]
>               "$ref": "./CommonDefinitions.json#/definitions/StringDictionary"

Usually 202 should just return headers only.
[https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#call-action-post-asynchronously](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#call-action-post-asynchronously)

---
Refers to: specification/web/resource-manager/Microsoft.Web/stable/2020-12-01/StaticSites.json:946 in 8f042f2. [](commit_id = 8f042f20612fdd7710def7f55d936c89b4e33cdc, deletion_comment = False)

**Comment by j5lim** [comment]
>             "description": "Asynchronous operation in progress.",

Need to set x-ms-long-running-operation

---
Refers to: specification/web/resource-manager/Microsoft.Web/stable/2020-12-01/StaticSites.json:1202 in 8f042f2. [](commit_id = 8f042f20612fdd7710def7f55d936c89b4e33cdc, deletion_comment = False)

**Comment by j5lim** [comment]
>               "$ref": "./CommonDefinitions.json#/definitions/RemotePrivateEndpointConnectionARMResource"

Should be empty body for 202

---
Refers to: specification/web/resource-manager/Microsoft.Web/stable/2020-12-01/StaticSites.json:2256 in 8f042f2. [](commit_id = 8f042f20612fdd7710def7f55d936c89b4e33cdc, deletion_comment = False)

**Comment by j5lim** [comment]
>           "202": {

Need to set x-ms-long-running-operation

---
Refers to: specification/web/resource-manager/Microsoft.Web/stable/2020-12-01/StaticSites.json:2595 in 8f042f2. [](commit_id = 8f042f20612fdd7710def7f55d936c89b4e33cdc, deletion_comment = False)

**Comment by j5lim** [comment]
>             "validationToken": {

Is this a secret? Then please annotate with x-ms-secret

---
Refers to: specification/web/resource-manager/Microsoft.Web/stable/2020-12-01/StaticSites.json:3001 in 8f042f2. [](commit_id = 8f042f20612fdd7710def7f55d936c89b4e33cdc, deletion_comment = False)

**Comment by j5lim** [comment]
>     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/sites/{name}/config/configreferences/connectionstrings": {

Does it return any secrets? If so, it should be POST.
[https://armwiki.azurewebsites.net/api_contracts/guidelines/rpc.html#rpc012-secret-property-semantics-are-followed](https://armwiki.azurewebsites.net/api_contracts/guidelines/rpc.html#rpc012-secret-property-semantics-are-followed)

---
Refers to: specification/web/resource-manager/Microsoft.Web/stable/2020-12-01/WebApps.json:1611 in 8f042f2. [](commit_id = 8f042f20612fdd7710def7f55d936c89b4e33cdc, deletion_comment = False)

**Comment by j5lim** [comment]
>             },

Is this a secret? Please mark all secrets in this file as x-ms-secret: true.

---
Refers to: specification/web/resource-manager/Microsoft.Web/stable/2020-12-01/WebApps.json:24424 in 8f042f2. [](commit_id = 8f042f20612fdd7710def7f55d936c89b4e33cdc, deletion_comment = False)

**Review by j5lim — CHANGES_REQUESTED** [review]
Left comments.

**Comment by pvbchak** [comment]
> > ```
> > "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/sites/{name}/config/configreferences/connectionstrings": {
> > ```
> 
> Does it return any secrets? If so, it should be POST.
> https://armwiki.azurewebsites.net/api_contracts/guidelines/rpc.html#rpc012-secret-property-semantics-are-followed
> 
> Refers to: specification/web/resource-manager/Microsoft.Web/stable/2020-12-01/WebApps.json:1611 in [8f042f2](https://github.com/Azure/azure-rest-api-specs/commit/8f042f20612fdd7710def7f55d936c89b4e33cdc). [](commit_id = [8f042f2](https://github.com/Azure/azure-rest-api-specs/commit/8f042f20612fdd7710def7f55d936c89b4e33cdc), deletion_comment = False)

It returns KeyVault References & does not contain any secrets

File: specification/web/resource-manager/Microsoft.DomainRegistration/stable/2020-12-01/examples/GetTopLevelDomain.json
Comment (pvbchak) [reply specification/web/resource-manager/Microsoft.DomainRegistration/stable/2020-12-01/examples/GetTopLevelDomain.json]: Fixed api-version in all examples

File: specification/web/resource-manager/Microsoft.CertificateRegistration/stable/2020-12-01/examples/ListOperations.json
Comment (pvbchak) [reply specification/web/resource-manager/Microsoft.CertificateRegistration/stable/2020-12-01/examples/ListOperations.json]: fixed api-version in all examples

File: specification/web/resource-manager/Microsoft.Web/stable/2020-12-01/Global.json
Comment (pvbchak) [reply specification/web/resource-manager/Microsoft.Web/stable/2020-12-01/Global.json:109]: removed.

File: specification/web/resource-manager/Microsoft.Web/stable/2020-12-01/Global.json
Comment (pvbchak) [reply specification/web/resource-manager/Microsoft.Web/stable/2020-12-01/Global.json]: removed.

File: specification/web/resource-manager/Microsoft.Web/stable/2020-12-01/Global.json
Comment (pvbchak) [reply specification/web/resource-manager/Microsoft.Web/stable/2020-12-01/Global.json]: removed from swagger

**Comment by miwebst** [comment]
@j5lim please take a look at the responses to your review. The completion of this PR is time sensitive so we would really appreciate your help on completing this.

**Comment by panchagnula** [comment]
@calvinsID can you verify all the swagger updates we need are already on this & sign-off?

**Review by naveedaz — APPROVED** [review]

**Review by calvinsID — APPROVED** [review]

**Comment by pilor** [comment]
>         "password": {

Missing x-ms-secret, ensure this isn't returned in GEt/PUT/PATCH responses

---
Refers to: specification/web/resource-manager/Microsoft.Web/stable/2020-12-01/WebApps.json:22609 in 7c95213. [](commit_id = 7c952134feefc2039ba2239c0294f2d1e46b3bf4, deletion_comment = False)

File: specification/web/resource-manager/Microsoft.Web/stable/2020-12-01/ResourceProvider.json
Comment (pvbchak) [reply specification/web/resource-manager/Microsoft.Web/stable/2020-12-01/ResourceProvider.json]: removed it in this release as it requires backend changes.

Comment (panchagnula) [reply specification/web/resource-manager/Microsoft.Web/stable/2020-12-01/ResourceProvider.json]: @ElleTojaroon FYI - we should get the code changes for these for Ant93.

**Comment by miwebst** [comment]
@j5lim - ARM has signed off and one of the tests above has an ssl error that appears to be flakey. Can you help rerun that particular test and help us check this in?

**Review by weidongxu-microsoft — APPROVED** [review]

**Comment by pvbchak** [comment]
> Left comments.

@j5lim Can you please look into updated review.

**Comment by j5lim** [comment]
> > Left comments.
> 
> @j5lim Can you please look into updated review.

@pvbchak It's signed off by ARM. Please work with the PR assignee to get it merged.

**Comment by miwebst** [comment]
@j5lim, I believe we have all the approvals, it seems we are blocked on the changes you requested. Is it possible for you to mark the reviewed changes as complete?

image

File: specification/web/resource-manager/Microsoft.Web/stable/2020-12-01/Certificates.json
Comment (j5lim) [reply specification/web/resource-manager/Microsoft.Web/stable/2020-12-01/Certificates.json:534]: It is still missing x-ms-secret.

---
In reply to: [593534876](https://github.com/Azure/azure-rest-api-specs/pull/13144#discussion_r593534876) [](ancestors = 593534876)

**Review by j5lim — APPROVED** [review]
Approved with suggestions.

**Comment by weidongxu-microsoft** [comment]
ARM approved, but with suggestions that might need update.

I would mark the PR good to merge. Please still check the suggestions, and let me know when you need it merged.

File: specification/web/resource-manager/Microsoft.Web/stable/2020-12-01/Certificates.json
Comment (pvbchak) [reply specification/web/resource-manager/Microsoft.Web/stable/2020-12-01/Certificates.json:534]: Fixed.

**Comment by pvbchak** [comment]
> ARM approved, but with suggestions that might need swagger update.
> 
> I would mark the PR good to merge. Please still check the suggestions, and let me know when you need it merged.

@weidongxu-microsoft I fixed it in latest iteration. Can you please merge once checks is all green.

File: custom-words.txt
Comment (nschonni) [inline custom-words.txt:2117]: Merge issue added back a bunch of typos

**Pull request merged**
